### ISSUE TITLE
[Link] Adding icon to link that opens new tab / window for accessiblity

### ISSUE THREAD

**Issue by bernietrinh** [telus/tds-core#1038]
## Description
We have an experience where we think it's better to open links in a new tab - a stepped flow with links (to external pages to apple/google) on the first step. This can potentially be an accessibility issue. Directly linking to those pages would otherwise mean users would be taken out of the flow and context.

## Recommendation
It is [recommended](https://www.w3.org/TR/WCAG20-TECHS/G200.html) that when links are opened to a new window, there is advance warning. Proposed solution is to provide an icon to act as the warning and have a standardized way of treating these type of links.

## Designs
![MTW V1 0_Suspend_Entry_Step_1 (3)](https://user-images.githubusercontent.com/6313985/56242085-7573d680-6065-11e9-91b5-34414b3f1d6a.png)

## Meta

- TDS component: `tds/core-link`
- Willing to develop solution: Yes
- Has workaround: No
- High impact: No
- Designer/Developer Pair: @bernietrinh @erhan-nascent

**Comment by donnavitan** [comment]
Hello @bernietrinh, we'll take this to design direction for review. It looks helpful.

I'm also going to include a note here from a previous discussion with @owestinTelus. While we don't have external link standards yet, please be consistent in your interim solution.

Thanks!

**Comment by donnavitan** [comment]
Hello @bernietrinh and @erhan-nascent,

We've been in discussion with design direction and this functionality would be a great enhancement to the link component. However, we do have a blocker and considerations to be addressed:
* TDS is currently in progress with updating our icon component to include an external window icon and a download icon. As soon as these 2 icons become available, they can be included in this enhancement.
* Enhancement should include a11y text to announce what the link is doing where necessary
* Limit enhancement so that someone using the link component can choose from a regular link, open in a new window link, and download file link. This method is preferred instead of allowing people to include any icon
* Add documentation providing guidance on when to use which link type

If this is something you can both work on and contribute, that would be greatly appreciated. Please let us know if you're able to take this on. We'll be happy to provide support in getting this design and development reviewed. Thanks again!

**Comment by donnavitan** [comment]
For references
* Download link example https://telusdigital.slack.com/archives/CDQ6P8VNU/p1556745185055100
* @jesdavpet mentioned that they implemented something like this in My TELUS with a document icon, perhaps you can reach out to him as well https://telusdigital.slack.com/archives/CDQ6P8VNU/p1556810004064800?thread_ts=1556745185.055100&cid=CDQ6P8VNU

**Comment by theetrain** [comment]
Hi @bernietrinh, the core team is looking into enhancing links as you've described; with an icon that represents navigating to an external site. Once we have designs ready, we'll reach out again to see if you have time to contribute.

**Comment by bernietrinh** [comment]
@theetrain sounds good, let me know and I should be able to help

**Comment by varunj90** [comment]
hey @bernietrinh this is currently in design and can be viewed here: https://telusdigital.atlassian.net/browse/TDS-1241

we will update you once updated.

**Comment by theetrain** [comment]
Small update:

As we roll out new icons, we are also designing ways to implement icons with buttons and links.

**Comment by theetrain** [comment]
Update:

This is blocked by our forthcoming Icon components. Once those are ready, we will take on icons within links and buttons.

**Comment by theetrain** [comment]
This icon is now available in the new Dependent icon group.

See:
* DSM Release v2.3.10
* React component: [@tds/core-interactive-icon@1.4.0](https://github.com/telus/tds-core/releases/tag/%40tds%2Fcore-interactive-icon%401.4.0)
* Documentation: https://tds.telus.com/components/index.html#!/Dependent
